From 8c430dd6b5c50b36ee8319f73754df0dfd625165 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 12 May 2024 22:46:04 +1100
Subject: [PATCH 16/31] add absl_die_if_null target

Paths without a/ b/ so conda-build applies under the real tree (see 0005).
Mirror the hunk into xla's system.absl.log.BUILD: tf_http_archive symlinks
from the label that resolves to that copy for absl/log.
---
 third_party/absl/system.absl.log.BUILD              | 16 ++++++++++++++++
 third_party/xla/third_party/absl/system.absl.log.BUILD | 16 ++++++++++++++++
 2 files changed, 32 insertions(+)

--- third_party/absl/system.absl.log.BUILD
+++ third_party/absl/system.absl.log.BUILD
@@ -3,6 +3,22 @@ load("@rules_cc//cc:defs.bzl", "cc_library")
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
+    name = "die_if_null",
+    linkopts = ["-labsl_die_if_null"],
+    deps = [
+        ":log",
+        "//absl/base:config",
+        "//absl/base:core_headers",
+        "//absl/strings",
+    ],
+)
+
+alias(
+    name = "absl_log",
+    actual = ":log",
+)
+
+cc_library(
     name = "log",
     linkopts = [
         "-labsl_vlog_config_internal",
--- third_party/xla/third_party/absl/system.absl.log.BUILD
+++ third_party/xla/third_party/absl/system.absl.log.BUILD
@@ -3,6 +3,22 @@ load("@rules_cc//cc:defs.bzl", "cc_library")
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
+    name = "die_if_null",
+    linkopts = ["-labsl_die_if_null"],
+    deps = [
+        ":log",
+        "//absl/base:config",
+        "//absl/base:core_headers",
+        "//absl/strings",
+    ],
+)
+
+alias(
+    name = "absl_log",
+    actual = ":log",
+)
+
+cc_library(
     name = "log",
     linkopts = [
         "-labsl_vlog_config_internal",
